### DISCUSSION TITLE
Wake from LE on BTN_LIGHT

### DISCUSSION THREAD

**Discussion by stealeks** [joeycastillo/Sensor-Watch#379] [Q&A] [answered]
Every night I am waking up and when I try to check out what time is it I am pushing Light button... but nothing happening, because watch is in LE mode... So.. I need to wake it first with Alarm button, and only then to push Light button.. kinda annoying..

Is it possible to wake watch via Light button instead of Alarm button? (First push - wake, second one LED as usual)

I tried to modify code.. but watch was forever sleeping :)
I guess I am blocked from hardware/PCB/wiring... Right?

**Comment by tahnok** [comment, marked as answer]
Correct, the current button is the only one that can wake the cpu from deep
sleep. You could make have the light turn on when you wake up?

**Reply by matheusmoreira** [reply]
> You could make have the light turn on when you wake up?
[…](https://github.com/joeycastillo/Sensor-Watch/discussions/379#discussioncomment-8722405)

I suggest the following feature:

 - Illuminate LED on LE wake up preference
   - One bit of movement state
   - LED lights up automatically on wake up if bit is set
 - Automatic LED management
   - Sunrise/sunset watch face background task
   - Memorizes the above bit's value
   - Sets the above bit automatically at sunset
   - Restores the above bit's previous value at sunrise
 - Configurable at runtime and at compile time

What do you think?

**Comment by stealeks** [comment]
Thank you! That's... a great idea! Will definitel try!
